src/System.IO.Pipelines.Extensions/UnownedBufferReader.cs

```diff
@@ -243,6 +243,11 @@ void IPipeReader.Complete(Exception exception)
             }
         }
 
+        public void OnWriterCompleted(Action<Exception, object> callback, object state)
+        {
+            throw new NotImplementedException();
```

Comment:
Hmmmm, we should fix this. Do a simple implementation based on task. Store a tcs internally and make that work.